src/VASL/build/module/map/VASLThread.java

```
@@ -538,9 +538,13 @@ private void setTargetFromRemoteEvent(Point eventPoint) {
 
         final Point p = mapMouseToMapCoordinates(eventPoint);
         adjustformagnification(p);
-        if (LOSMap == null) {return;} // error handling
-        if (p == null || !LOSMap.onMap(p.x, p.y)) {return;} // error handling
-        target = LOSMap.gridToHex(p.x, p.y).getNearestLocation(p.x, p.y);
+        if (LOSMap == null) {return;} // error handling - no VASLMap
+        if (p == null || !LOSMap.onMap(p.x, p.y)) {return;} // error handling - no point or point not on map
+        try { // error handling - can't find a location from point; parts of hexes on map edge?
+            target = LOSMap.gridToHex(p.x, p.y).getNearestLocation(p.x, p.y);
+        }catch (Exception e) {
```

There are hardly any circumstances where you should catch `Exception`. Whatever you're trying to catch should have a more specific type. Catching `Exception` will mask exceptions which are not the type you're intending to trap.

---

I have pushed a change making it an NullPointException. Thanks for the advice. I am starting 672 development by trying to clear up all the issues transferred from VASSAL many of which cause crashes and I will probably use try/catch in several places. Will use specific exceptions where possible.

---

That's also wrong. You should never catch an NPE. Instead, check for `null` before using the result.

---

Have googled this and begin to understand why it is not a good idea to catch an NPE. I will proceed as you suggest. 

Lots of learning today!